### ISSUE TITLE
Classes not being included in build when used in field 

### ISSUE THREAD

**Issue by devyndamonster** [H3VR-Modding/Atlas#1]
When building my map with this a monobehaviour that uses a non-monobehaviour in one of it's fields, the non-monobehaviour does not get included in the build. Below is my setup on a gameobject.

![teams](https://user-images.githubusercontent.com/6070388/139596281-702ec28e-76b3-46a7-8669-19a57c9aff24.PNG)

The elements in the `Teams` list are seperate classes, shown below

![team2](https://user-images.githubusercontent.com/6070388/139596287-f925f5a8-fa75-49d9-ac69-23dfab9f5d3f.PNG)

When running the game, the following code is run

![team3](https://user-images.githubusercontent.com/6070388/139596288-996ef141-3d2e-4b6c-9635-76d3fed16682.PNG)

But in game it's like the elements of the `Teams` list where never brought in, and so the list is seen as empty. You can see the Team count below is marked as 0

```
[Error  : Unity Log] Something bad happened!
[Error  : Unity Log] Team Count: 0, Hill Count: 1
[Error  : Unity Log] Error :
System.ArgumentOutOfRangeException: Argument is out of range.
Parameter name: index
  at System.Collections.Generic.List`1[KOTH.KOTHTeam].get_Item (Int32 index) [0x00000] in <filename unknown>:0 
  at KOTH.KOTHManager.SpawnSosig (Int32 team) [0x00000] in <filename unknown>:0 
```

**Comment by devyndamonster** [comment]
For reference, the index I access teams with is always within range. But that shouldn't even be relevant as you can see the team count printed is 0 anyway

**Comment by nrgill28** [comment]
Apparently this is a [known Unity issue](https://issuetracker.unity3d.com/issues/assetbundle-is-not-loaded-correctly-when-they-reference-a-script-in-custom-dll-which-contains-system-dot-serializable-in-the-build) that they have chosen to not fix (not that it would help us on 5.6 anyway). Serializable classes in external assemblies just do not deserialize correctly, you'll have to workaround it by using monobehaviours / scriptableobjects. There's a discussion on the Unity forums here about it: https://forum.unity.com/threads/serialization-lost-when-loading-asset-bundles-in-runtime.587431/

**Comment by devyndamonster** [comment]
Lmao, good stuff Unity. Thanks for checking anyway!
